pandas/tests/io/excel/test_readers.py

```diff
@@ -455,12 +460,17 @@ def test_reader_special_dtypes(self, request, read_ext):
 
         no_convert_float = float_expected.copy()
         no_convert_float["StrCol"] = no_convert_float["StrCol"].apply(str)
-        actual = pd.read_excel(
-            basename + read_ext,
-            sheet_name="Sheet1",
-            convert_float=False,
-            converters={"StrCol": str},
-        )
+        with tm.assert_produces_warning(
+            FutureWarning,
+            match="convert_float is deprecated",
+            raise_on_extra_warnings=False,
```

Comment:
Aside: This was added because of the stacklevel issue, is that right? If so, it seems like incorrect behavior in assert_produces_warning, this should still fail is the warning were looking for isn't raised at the correct stacklevel.

Reply:
I added the `raise_on_extra_warnings` because Linux_py37_IO (ci/deps/actions-37-db.yaml) raises a PendingDeprecationWarning from within xlrd during two cases of test_readers.test_reader_special_dtypes:
https://github.com/pandas-dev/pandas/pull/41176/checks?check_run_id=2458552180

All the other test environments came back green. Should I add a comment to the test code to explain?

Reply:
Ah, I see, thanks. Yes - I think a comment would be good here.

Reply:
Comment has been added, thanks.